awscli/customizations/cloudfront.py

```diff
@@ -50,14 +86,83 @@ def __init__(self):
+                             exclusive_to='distribution-config', **kwargs):
+    argument_table[exclusive_to].required = False
+    for arg in args:
+        argument_table[arg.name] = arg
+
+
+class ExclusiveArgument(CustomArgument):
+    DOC = '%s This argument and --%s are mututally exclusive.'
+
+    def __init__(self, name, exclusive_to, help_text=''):
+        super(ExclusiveArgument, self).__init__(
+            name, help_text=self.DOC % (help_text, exclusive_to))
+
+    def add_to_params(self, parameters, value):
+        if value is not None:
+            parameters[self.name] = value
+
+
+def _create_distribution(call_parameters, **kwargs):
+    if call_parameters and 'DistributionConfig' not in call_parameters:
+        domain_name = call_parameters.pop('origin-domain-name', '')
+        origin_id = unique_string(prefix=domain_name)
+        item = {"Id": origin_id, "DomainName": domain_name, "OriginPath": ''}
+        if item['DomainName'].endswith('.s3.amazonaws.com'):
+            # We do not need to detect '.s3[\w-].amazonaws.com' as S3 buckets,
+            # because CloudFront treats GovCloud S3 buckets as custom domain.
+            # http://docs.aws.amazon.com/govcloud-us/latest/UserGuide/setting-up-cloudfront.html
+            item["S3OriginConfig"] = {"OriginAccessIdentity": ""}
+        else:
+            item["CustomOriginConfig"] = {
+                'HTTPPort': 80, 'HTTPSPort': 443,
+                'OriginProtocolPolicy': 'http-only'}
+        call_parameters['DistributionConfig'] = {
+            "CallerReference": unique_string(),
+            "Origins": {"Quantity": 1, "Items": [item]},
+            "DefaultCacheBehavior": {
+                "TargetOriginId": origin_id,
+                "ForwardedValues": {
+                    "QueryString": False,
+                    "Cookies": {"Forward": "none"},
+                },
+                "TrustedSigners": {
+                    "Enabled": True,
+                    "Quantity": 0
+                },
+                "ViewerProtocolPolicy": "allow-all",
+                "MinTTL": 0
+            },
+            "Enabled": True,
+            "Comment": "",
+        }
+        if 'default-root-object' in call_parameters:
+            call_parameters['DistributionConfig']['DefaultRootObject'] = \
+                call_parameters.pop('default-root-object')
+
+
+def _update_distribution(parsed_globals, call_parameters, **kwargs):
+    if call_parameters and 'DistributionConfig' not in call_parameters:
+        client = Session().create_client(
+            'cloudfront', region_name=parsed_globals.region,
+            endpoint_url=parsed_globals.endpoint_url,
+            verify=parsed_globals.verify_ssl)
+        response = client.get_distribution_config(Id=call_parameters['Id'])
+        call_parameters['IfMatch'] = response['ETag']
+        call_parameters['DistributionConfig'] = response['DistributionConfig']
+        if 'default-root-object' in call_parameters:
```

Comment:
Why would this be in the ``call_parameters``? It should only be in the ``parsed_args``. We should not be putting parameters into the ``call_parameters`` in the first place that has no chance to be used by the client.